### ISSUE TITLE
Optimize dependancies solving with C ecosystem library

### ISSUE THREAD

**Issue by dev590t** [pdm-project/pdm#1142]
## Is your feature request related to a problem? Please describe.

Dependancies solving is a high performance computing task. I'm never convinced that Python is the good language for such task. For some project, PDM has take more one hour to try to solve the dependancies, and it did not succeed.

## Describe the solution you'd like

I appreciate there is a file pdm.lock, that allow developer to solve dependancies with other high performance tool, where PDM fails. After we can just use `pdm install` to install dependancies.

But I think that could be better if pdm can use directly some high performance dependancie solver from C ecosystem.

**Comment by pawamoy** [comment]
This has been discussed on the Poetry tracker I think: solving Python dependencies is not a CPU- but a network-bound task, because the data isn't provided by the index since most of the packages still use dynamic setups. Using C won't help at all. Let me try to find the answer from, I think, Sébastien Eustache :slightly_smiling_face:

**Comment by pawamoy** [comment]
By the way

> that allow developer to solve dependancies with other high performance tool, where PDM fails.

any examples of such tools? I'd be curious to see and try them!

**Comment by frostming** [comment]
Yes, most of the time the solver is downloading distributions, extracting them, and reading the metadata. It is a heavy IO-bound task. C implementations don't help in this scenario.

AFAIK, conda is using [mamba](https://github.com/mamba-org/mamba) with a carefully designed package server, which calculates and caches the dependency map on the server. But I don't think we can easily adopt it.
